core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1350,7 +1352,13 @@ class Log(@volatile var dir: File,
       }
 
       // check the validity of the message by checking CRC
-      batch.ensureValid()
+      try {
```

Comment:
Being a bit paranoid here: we can use  the`boolean batch.isValid()` function so that we can replace the try-catch here.

Reply:
ack